src/chains/ethereum/block/src/pending-block.ts

```diff
@@ -0,0 +1,60 @@
+import { Data, Quantity } from "@ganache/utils";
+import type Common from "@ethereumjs/common";
+import { GanacheTrie } from "@ganache/ethereum-utils";
+import { Block } from "./block";
+
+export class PendingBlock extends Block {
+  /**
+   * Used to override the `stateRoot` value of the return value of the `toJSON`
+   * function.
+   */
+  #toJSONStateRootOverride = Data.from("0x", 32);
+  /**
+   * Used to override the `hash` value of the return value of the `toJSON`
+   * function.
+   */
+  #toJSONHashOverride = Quantity.Empty;
+  /**
+   * Used to store the trie that was used when mining the block.
+   */
+  #trie: GanacheTrie;
+
+  constructor(common: Common, trie: GanacheTrie) {
+    super(null, common);
+    this.#trie = trie;
+  }
+
+  /**
+   * Calls the `Block` class' `toJSON` function, then overwrites the `stateRoot`
+   * and `hash` properties of the result with the pending block's `stateRoot`
+   * and `hash`.
+   * @param includeFullTransactions
+   * @returns The pending block's data in JSON format.
+   */
+  toJSON<IncludeTransactions extends boolean>(
+    includeFullTransactions: IncludeTransactions
+  ) {
+    const json = super.toJSON(includeFullTransactions);
+    json.stateRoot = this.#toJSONStateRootOverride;
+    json.hash = this.#toJSONHashOverride;
+
+    return json;
+  }
```

Comment:
We are overriding the toJSON function because we still _need_ the actual `stateRoot` and `hash` values of the pending block for use cases that traverse the trie. For example, if we were to override the `stateRoot`, this function in your `eth_getProof` RPC method wouldn't work: 
```
 await stateManagerCopy.setStateRoot(
      targetBlock.header.stateRoot.toBuffer()
    );
```

However, geth returns empty data for the `toJSON` representation of the pending block, hence this "half-measure" override of the data.